### PR TITLE
feat: add project name and credit class name to tables

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1072 regen-network/regen-registry#1191 regen-network/regen-registry#1225

- add the project name and credit class name columns to all data tables.
- update hooks and API calls to project name and credit class name.
- add common style for an ellipsis column with approx ~10 char.acters
- prevent unnecessary requests from `useBasketTokens` hook.
- filter credit class column on some pages

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-1261--regen-registry.netlify.app/ecocredits/dashboard
2. check new project name + credit class name

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can test this story: https://deploy-preview-1261--regen-registry.netlify.app/ecocredits/dashboard
all tables should now have project name + credit class name displayed.

File: web-registry/src/styles/table.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { SxProps } from '@mui/material';
+
+export const ELLIPSIS_COLUMN_WIDTH = '125px';
+
+export const tableStyles = {
```
Comment (flagrede) [inline web-registry/src/styles/table.ts:5]: Created `src/styles` so we can start to mutualize some common CSS written as `sx` prop.

**Comment by erikalogie** [comment]
This looks great! 

I will add a follow up task to make it so that clicking the "..." will open the column to reveal the whole name.

**Comment by flagrede** [comment]
> This looks great!
> 
> I will add a follow up task to make it so that clicking the "..." will open the column to reveal the whole name.

@erikalogie About this, I wasn't how it would work since these fields are already used as links.

**Comment by erikalogie** [comment]
> > This looks great!
> > I will add a follow up task to make it so that clicking the "..." will open the column to reveal the whole name.
> 
> @erikalogie About this, I wasn't how it would work since these fields are already used as links.

Totally, let's tackle this separately!

**Review by victordelval — APPROVED** [review]

**Review by mhagel — APPROVED** [review]

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**